src/Features/Core/Portable/RemoveUnusedMembers/AbstractRemoveUnusedMembersDiagnosticAnalyzer.cs

```diff
@@ -375,14 +402,26 @@ private void OnSymbolEnd(SymbolAnalysisContext symbolEndContext, bool hasInvalid
                                 continue;
                             }
 
+                            var messageFormat = rule.MessageFormat;
+                            if (rule == s_removeUnreadMembersRule &&
+                                member is IMethodSymbol)
+                            {
+                                // IDE0052 has a different message for method symbols.
+                                messageFormat = FeaturesResources.Private_method_0_can_be_removed_as_it_is_never_invoked;
+                            }
```

Comment:
this feels a bit weird.  the rule has a format... but then we'd just overriding it.  It makes it seem as if the message isn't inherent to the rule.

Either, pull all the message choosing up to here, or make it so that we query the rule for the appropriate message, apssing in whatever data is needs to make that decision (i.e. the member in this case).

Reply:
Sure, will refactor